lib/takeaway.rb

```diff
@@ -0,0 +1,51 @@
+require_relative 'Menu'
+require("bundler")
+Bundler.require()
+
+
+class Takeaway
+  attr_reader :basket, :result,:menu
+
+  def initialize
+    @basket = []
+    @menu = nil
+    @total = 0
+  end
+
+  def see(menu)
+    @menu = menu.list
+  end
+
+  def select(item,quantity=1)
+    @result = @menu.find { |x| x[item] }
+    @basket.concat([@result]* quantity)
+    end
+
+  def total
+    @basket.each { |x|
+      x.each { |key, value|
+      @total = @total + value }}
+      @total
+  end
+
+  def is_amount_correct(total)
```

Comment:
Nice! Could you think about a way of testing this method to check it gives the right response with the time? Perhaps Time.new could be passed in as a dependency to the object?

Reply:
Hi Lindsey, 

Thank you for your suggestion. I did some research and was thinking I could test this method with something like this: 

allow(Time).to receive(:now).and_return(@time_now)

I am going to try this with my Chitter challenge as I used Time.now there in a method as well :)